server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/TermsValuesSourceBuilder.java

```diff
@@ -84,6 +86,13 @@ protected CompositeValuesSourceConfig innerBuild(SearchContext context, ValuesSo
             vs = ValuesSource.Numeric.EMPTY;
         }
         final MappedFieldType fieldType = config.fieldContext() != null ? config.fieldContext().fieldType() : null;
-        return new CompositeValuesSourceConfig(name, fieldType, vs, config.format(), order());
+        final DocValueFormat format;
+        if (format() == null && fieldType instanceof DateFieldMapper.DateFieldType) {
+            // defaults to the raw format on date fields (preserve timestamp as longs).
+            format = DocValueFormat.RAW;
```

Comment:
why shouldn't we use the format from the mapping?

Reply:
That's a good question. In 6.2 we always return the raw value and there is no distinction in this aggregation to return `getKey` or `getKeyAsString` since the returned object is a map. So I think it's preferable to return the raw value unless a `format` has been specified.

Reply:
I see. for the record we'll probably need to change this anyway one day for #18518